Lib/ufo2ft/postProcessor.py

```diff
@@ -164,7 +165,10 @@ def _rename_glyphs_from_ufo(self):
         """Rename glyphs using ufo.lib.public.postscriptNames in UFO."""
         rename_map = self._build_production_names()
 
-        otf = self.otf
+        # NOTE: We must reload the font here, as loaded tables will keep using
+        # old glyph names. Compilation will convert them to glyph indices,
+        # which we can then rename by modifying the `post` or `CFF*` tables.
+        otf = self.otf = _reloadFont(self.otf)
```

Comment:
May be renaming should be done after compiling the font? Compile, open the font and only modify the `post` table (which I suppose should load the other heavy tables). Unless this is already what is happening here.

Reply:
Cosimo told me to do it only when we truly want to rename glyphs, i.e. inside the rename function. For some reason this still creates mismatches with some variable font TTX dumps so I have to reload when returning from `compileVariable*`, too 😭 I'm sure there are better data flows (i.e. rename right from the start and give a mapping to feaLib) but I dunno 😢

Reply:
the ufo2ft integration tests only check the TTX dump of the TTFont that comes out ufo2ft's own compile functions. Usually that's going to be saved (compiled) to disk. The compile operation will also take care dropping the glyph names if post was set to format 3.0 (the tables which had been previously decompiled will be recompiled using the old names, post 3.0 will no longer contain the glyph names, such that upon reopening the same font the glyph names will be made up by fonttools from cmap or AGL names).

In general the problem when renaming or dropping glyph names is that there may already be tables that had been loaded using the old glyph names, and thus need to be compiled using the old names. Which is why renaming has to happen as first thing before any other table has been loaded, or else one needs to first reload the font (i.e compile all tables that had been loaded using old names, or simply copy the ones that haven't been loaded and are still binary blobs in the SFNTReader), and then they can change the names on the new TTFont instance.

Reply:
> Unless this is already what is happening here

well, the rename operation is the last thing that the postProcessor does, and the post-processor already is supposed to be "at the end", so it is kind of already happening

Reply:
Problem I see is that we are compiling masters, each getting their postProcessor running, and then we merge those masters into a varfont. I think we should only run postProcessor on the varfont, not on the masters.

Reply:
> I think we should only run postProcessor on the varfont, not on the masters.

indeed, the postProcessor is not actually doing anything when the `compileInterpolatable*FromDS` funcs are being called from the respective `compileVariable{TTF,CFF2}` functions, we are deliberately turning off subroutinization (for CFF only) as well as glyph renaming (via useProductionNames=False).
We could make so that we skip the whole post-processor step when building the interpolatable master TTFs/OTFs and only run at the very end of the (static/variable) pipeline. Then we could keep a single reloadFont in the postProcessor, just before we rename glyphs or immediatley after we drop them (setting post 3.0), which should make sure the TTX dumps contain the expected names.
To keep it simple, we could change the `compileInterpolatable*FromDS` to never call the postProcessor, basically ignoring the `userProductionNames` parameter. The problem is those same functions are not only used internally by `compileVariable*`, but are also used by fontmake for the `-o {o,t}tf-interpolatable` targets and currently they do respect the `userProductionNames` parameter (even though it doesn't make much sense to do glyph renaming on these intermediate build products). Maybe one can argue that the whole point of the interpolatable binary master is to build variable fonts so ignoring the glyph renaming process should be the default and only behavior.